media/server/gstplayer/source/GstGenericPlayer.cpp

```diff
@@ -1523,14 +1653,73 @@ void GstGenericPlayer::processAudioGap(int64_t position, uint32_t duration, int6
     }
 }
 
+void GstGenericPlayer::setBufferingLimit(uint32_t limitBufferingMs)
+{
+    if (m_workerThread)
+    {
+        m_workerThread->enqueueTask(m_taskFactory->createSetBufferingLimit(m_context, *this, limitBufferingMs));
+    }
+}
+
+bool GstGenericPlayer::getBufferingLimit(uint32_t &limitBufferingMs)
+{
+    bool returnValue{false};
+    GstElement *decoder = getDecoder(MediaSourceType::AUDIO);
+    if (decoder && m_glibWrapper->gObjectClassFindProperty(G_OBJECT_GET_CLASS(decoder), "limit-buffering-ms"))
+    {
+        m_glibWrapper->gObjectGet(decoder, "limit-buffering-ms", &limitBufferingMs, nullptr);
+        returnValue = true;
+    }
+    else if (m_context.pendingBufferingLimit.has_value())
+    {
+        RIALTO_SERVER_LOG_DEBUG("Returning queued value");
+        limitBufferingMs = m_context.pendingBufferingLimit.value();
+        returnValue = true;
+    }
+    else
+    {
+        RIALTO_SERVER_LOG_ERROR("buffering limit not supported in decoder '%s'",
+                                (decoder ? GST_ELEMENT_NAME(decoder) : "null"));
+    }
+
+    if (decoder)
+        m_gstWrapper->gstObjectUnref(GST_OBJECT(decoder));
+
+    return returnValue;
+}
+
+void GstGenericPlayer::setUseBuffering(bool useBuffering)
+{
+    if (m_workerThread)
+    {
+        m_workerThread->enqueueTask(m_taskFactory->createSetUseBuffering(m_context, *this, useBuffering));
+    }
+}
+
+bool GstGenericPlayer::getUseBuffering(bool &useBuffering)
+{
+    if (m_context.playbackGroup.m_curAudioDecodeBin)
```

Comment:
Using m_context.playbackGroup.m_curAudioDecodeBin on the main thread may not be thread safe

Reply:
m_curAudioDecodeBin changed to atomic. Additionally, property mutex introduced to protect properties in get/set methods